### PR TITLE
♻️ Refactor/liquidations initiator

### PR DESCRIPTION
**Pull request by Thomas-Smets**

### CODE REVIEW

File: src/test/9_Liquidator.t.sol [hunk 1]
```diff
@@ -88,82 +85,246 @@ contract LiquidatorTest is DeployArcadiaVaults {
...
+                            AUCTION LOGIC
+    ///////////////////////////////////////////////////////////////*/
+
+    function testRevert_startAuction_NonVault(
+        address unprivilegedAddress_,
+        address liquidationInitiator_,
+        uint128 openDebt
+    ) public {
+        vm.assume(unprivilegedAddress_ != address(proxy));
+        vm.assume(unprivilegedAddress_ != address(vault));
```
Comment (j-vp) [inline src/test/9_Liquidator.t.sol:204]: this should be a valid address here (should revert)

File: src/Liquidator.sol [hunk 2]
```diff
@@ -115,58 +108,72 @@ contract Liquidator is Ownable {
...
      */
-    function getPriceOfVault(address vaultAddress, uint256 life)
+    function getPriceOfVault(address vaultAddress)
         public
         view
-        returns (uint256 totalPrice, uint8 baseCurrencyOfVault, bool forSale)
+        returns (uint256 totalPrice, address baseCurrency, bool forSale)
     {
-        forSale = !(auctionInfo[vaultAddress][life].stopped) && auctionInfo[vaultAddress][life].startBlock > 0;
+        forSale = auctionInformation[vaultAddress].inAuction && auctionInformation[vaultAddress].startBlock > 0;
```
[the remark this answers is not in the document]
Comment (Thomas-Smets) [reply src/Liquidator.sol:168]: No, already fixed in https://github.com/arcadia-finance/arcadia-vaults/tree/refactor/liquidations-buy-vault, where we only check on  inAuction == true

File: src/Liquidator.sol [hunk 3]
```diff
@@ -177,193 +184,61 @@ contract Liquidator is Ownable {
...
-        (uint256 priceOfVault,, bool forSale) = getPriceOfVault(vaultAddress, life);
+    function buyVault(address vaultAddress) public {
+        (uint256 priceOfVault,, bool forSale) = getPriceOfVault(vaultAddress);
 
         require(forSale, "LQ_BV: Not for sale");
 
-        address lendingPool = IVault(vaultAddress).trustedProtocol();
-        address asset = ILendingPool(lendingPool).asset();
+        address trustedCreditor = auctionInformation[vaultAddress].trustedCreditor;
+        address asset = auctionInformation[vaultAddress].baseCurrency;
```
[the remark this answers is not in the document]
Comment (Thomas-Smets) [reply src/Liquidator.sol:201]: See https://github.com/arcadia-finance/arcadia-vaults/tree/refactor/liquidations-buy-vault

File: src/Liquidator.sol [hunk 4]
```diff
@@ -177,193 +184,61 @@ contract Liquidator is Ownable {
...
-        uint256 openDebt = auctionInfo[vaultAddress][life].openDebt;
-        claimRatios memory ratios = claimRatio;
-        uint256 keeperReward = openDebt * ratios.liquidationKeeper / 100;
+        uint256 openDebt = auctionInformation[vaultAddress].openDebt;
+        ClaimRatios memory ratios = claimRatios;
+        uint256 keeperReward = openDebt * ratios.liquidationInitiator / 100;
 
         if (priceOfVault < openDebt + keeperReward) {
             uint256 default_ = openDebt + keeperReward - priceOfVault;
             uint256 deficit = priceOfVault < keeperReward ? keeperReward - openDebt : 0;
```
[the remark this answers is not in the document]
Comment (Thomas-Smets) [reply src/Liquidator.sol:211]: See https://github.com/arcadia-finance/arcadia-vaults/tree/refactor/liquidations-buy-vault

File: src/Liquidator.sol [hunk 5]
```diff
@@ -177,193 +184,61 @@ contract Liquidator is Ownable {
...
             //uint256 protocolReward = openDebt * ratios.protocol / 100;
-            //uint256 surplus = priceOfVault - auctionInfo[vaultAddress][life].openDebt;
+            //uint256 surplus = priceOfVault - auctionInformation[vaultAddress][life].openDebt;
             //protocolReward = surplus > protocolReward ? protocolReward  : surplus;
         }
 
-        auctionInfo[vaultAddress][life].assetPaid = uint128(priceOfVault);
-        auctionInfo[vaultAddress][life].stopped = true;
+        auctionInformation[vaultAddress].inAuction = false;
+        //ToDo: set all auction information to 0?
```
[the remark this answers is not in the document]
Comment (Thomas-Smets) [reply src/Liquidator.sol:225]: See https://github.com/arcadia-finance/arcadia-vaults/tree/refactor/liquidations-buy-vault

File: src/test/8_Vault.t.sol [hunk 6]
```diff
@@ -786,41 +786,34 @@ contract LiquidationLogicTest is vaultTests {
...
+        stdstore.target(address(debt)).sig(debt.realisedDebt.selector).checked_write(usedMargin);
+        stdstore.target(address(debt)).sig(debt.balanceOf.selector).with_key(address(vault_)).checked_write(usedMargin);
 
-        vm.expectRevert("V: You are not the factory");
-        vault_.liquidateVault(liquidationKeeper);
+        vm.prank(address(factory));
+        address liquidator_ = vault_.liquidateVault(liquidationInitiator);
 
-        assertEq(vault_.owner(), vaultOwner);
+        assertEq(liquidator_, address(liquidator));
```
[the remark this answers is not in the document]
Comment (Thomas-Smets) [reply src/test/8_Vault.t.sol:816]: That's not for this unit test, we already test it in testSuccess_liquidate() in 7_Factory.t.sol

**Pull request merged**